<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" /><meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  <title>Running HBA-DEALS &mdash; hbadeals 1.0.0 documentation</title>
      <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
      <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
      <link rel="stylesheet" href="_static/style.css" type="text/css" />
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
        <script src="_static/jquery.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/doctools.js"></script>
        <script src="_static/sphinx_highlight.js"></script>
    <script src="_static/js/theme.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="prev" title="Inferring transcript readcounts" href="transcriptome.html" /> 
</head>

<body class="wy-body-for-nav"> 
  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >

          
          
          <a href="index.html" class="icon icon-home">
            hbadeals
          </a>
              <div class="version">
                1.0
              </div>
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>
        </div><div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption" role="heading"><span class="caption-text">Contents:</span></p>
<ul class="current">
<li class="toctree-l1"><a class="reference internal" href="setup.html">Setting up HBA-DEALS</a></li>
<li class="toctree-l1"><a class="reference internal" href="cohort.html">Cohort RNA-seq data</a></li>
<li class="toctree-l1"><a class="reference internal" href="transcriptome.html">Inferring transcript readcounts</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Running HBA-DEALS</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#hba-deals-arguments">HBA-DEALS arguments</a></li>
<li class="toctree-l2"><a class="reference internal" href="#output-format">Output format</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap"><nav class="wy-nav-top" aria-label="Mobile navigation menu" >
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">hbadeals</a>
      </nav>

      <div class="wy-nav-content">
        <div class="rst-content">
          <div role="navigation" aria-label="Page navigation">
  <ul class="wy-breadcrumbs">
      <li><a href="index.html" class="icon icon-home" aria-label="Home"></a></li>
      <li class="breadcrumb-item active">Running HBA-DEALS</li>
      <li class="wy-breadcrumbs-aside">
            <a href="_sources/hbadeals.rst.txt" rel="nofollow"> View page source</a>
      </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
             
  <section id="running-hba-deals">
<span id="rsthbadeals"></span><h1>Running HBA-DEALS<a class="headerlink" href="#running-hba-deals" title="Permalink to this heading"></a></h1>
<p>We will now read the expected counts from the rsem directory and create a count matrix, which we will pass as input to HBA-DEALS.</p>
<p>The following code creates the counts matrix</p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="n">countsData</span><span class="o">=</span><span class="kc">NULL</span><span class="w">  </span><span class="c1">#The counts matrix</span>
<span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">()</span><span class="w">   </span><span class="c1">#The sample labels, 1 for control and 2 for case</span>
<span class="nf">for </span><span class="p">(</span><span class="n">srr</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;SRR7236472&#39;</span><span class="p">,</span><span class="s">&#39;SRR7236473&#39;</span><span class="p">,</span><span class="s">&#39;SRR7236474&#39;</span><span class="p">,</span><span class="s">&#39;SRR7236475&#39;</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">    </span><span class="nf">if </span><span class="p">(</span><span class="nf">is.null</span><span class="p">(</span><span class="n">countsData</span><span class="p">))</span>
<span class="w">        </span><span class="n">countsData</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;rsem/&#39;</span><span class="p">,</span><span class="n">srr</span><span class="p">,</span><span class="s">&#39;.isoforms.results&#39;</span><span class="p">),</span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)[</span><span class="nf">c</span><span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="m">1</span><span class="p">)]</span>
<span class="w">    </span><span class="n">next.file</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;rsem/&#39;</span><span class="p">,</span><span class="n">srr</span><span class="p">,</span><span class="s">&#39;.isoforms.results&#39;</span><span class="p">),</span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">    </span><span class="n">countsData</span><span class="o">=</span><span class="nf">cbind</span><span class="p">(</span><span class="n">countsData</span><span class="p">,</span><span class="n">next.file</span><span class="o">$</span><span class="n">expected_count</span><span class="p">)</span>
<span class="w">   </span><span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span><span class="m">1</span><span class="p">)</span>
<span class="p">}</span>
<span class="nf">for </span><span class="p">(</span><span class="n">srr</span><span class="w"> </span><span class="n">in</span><span class="w"> </span><span class="nf">c</span><span class="p">(</span><span class="s">&#39;SRR7236480&#39;</span><span class="p">,</span><span class="s">&#39;SRR7236481&#39;</span><span class="p">,</span><span class="s">&#39;SRR7236482&#39;</span><span class="p">,</span><span class="s">&#39;SRR7236483&#39;</span><span class="p">))</span><span class="w"> </span><span class="p">{</span>
<span class="w">   </span><span class="n">next.file</span><span class="o">=</span><span class="nf">read.table</span><span class="p">(</span><span class="nf">paste0</span><span class="p">(</span><span class="s">&#39;rsem/&#39;</span><span class="p">,</span><span class="n">srr</span><span class="p">,</span><span class="s">&#39;.isoforms.results&#39;</span><span class="p">),</span><span class="n">header</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">)</span>
<span class="w">   </span><span class="n">countsData</span><span class="o">=</span><span class="nf">cbind</span><span class="p">(</span><span class="n">countsData</span><span class="p">,</span><span class="n">next.file</span><span class="o">$</span><span class="n">expected_count</span><span class="p">)</span>
<span class="w">   </span><span class="n">labels</span><span class="o">=</span><span class="nf">c</span><span class="p">(</span><span class="n">labels</span><span class="p">,</span><span class="m">2</span><span class="p">)</span>
<span class="p">}</span>
<span class="n">countsData</span><span class="o">=</span><span class="n">countsData</span><span class="p">[</span><span class="nf">rowSums</span><span class="p">(</span><span class="n">countsData</span><span class="p">[,</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">)]</span><span class="o">&gt;</span><span class="m">0</span><span class="p">)</span><span class="o">&gt;=</span><span class="nf">ncol</span><span class="p">(</span><span class="n">countsData</span><span class="p">)</span><span class="m">-2</span><span class="p">,]</span><span class="w"> </span><span class="c1">#remove low-count isoforms</span>
<span class="n">num.iso</span><span class="o">=</span><span class="nf">unlist</span><span class="p">(</span><span class="nf">lapply</span><span class="p">(</span><span class="n">countsData</span><span class="o">$</span><span class="n">gene_id</span><span class="p">,</span><span class="nf">function</span><span class="p">(</span><span class="n">x</span><span class="p">){</span><span class="nf">sum</span><span class="p">(</span><span class="n">countsData</span><span class="o">$</span><span class="n">gene_id</span><span class="w"> </span><span class="o">%in%</span><span class="w"> </span><span class="n">x</span><span class="p">)}))</span>
<span class="n">countsData</span><span class="o">=</span><span class="n">countsData</span><span class="p">[</span><span class="n">num.iso</span><span class="o">&gt;</span><span class="m">1</span><span class="p">,]</span><span class="w">  </span><span class="c1">#keep only genes with at least two isoforms</span>
</pre></div>
</div>
<p>The following code performs the HBA-DEALS analysis.</p>
<div class="highlight-R notranslate"><div class="highlight"><pre><span></span><span class="nf">library</span><span class="p">(</span><span class="n">hbadeals</span><span class="p">)</span>
<span class="nf">library</span><span class="p">(</span><span class="n">edgeR</span><span class="p">)</span>
<span class="n">res</span><span class="o">=</span><span class="nf">hbadeals</span><span class="p">(</span><span class="n">countsData</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">countsData</span><span class="p">,</span><span class="n">labels</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="n">labels</span><span class="p">,</span><span class="n">n.cores</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="m">32</span><span class="p">,</span><span class="n">isoform.level</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span><span class="n">mcmc.iter</span><span class="o">=</span><span class="m">100000</span><span class="p">,</span><span class="n">mcmc.warmup</span><span class="o">=</span><span class="m">10000</span><span class="p">,</span><span class="n">mtc</span><span class="o">=</span><span class="kc">TRUE</span><span class="p">,</span>
<span class="w">         </span><span class="n">lib.size</span><span class="o">=</span><span class="nf">colSums</span><span class="p">(</span><span class="n">countsData</span><span class="p">[,</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">)])</span><span class="o">*</span><span class="nf">calcNormFactors</span><span class="p">(</span><span class="nf">as.matrix</span><span class="p">(</span><span class="n">countsData</span><span class="p">[,</span><span class="o">-</span><span class="nf">c</span><span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="m">2</span><span class="p">)]),</span><span class="n">method</span><span class="o">=</span><span class="s">&#39;TMM&#39;</span><span class="p">))</span>
<span class="nf">write.table</span><span class="p">(</span><span class="n">res</span><span class="p">,</span><span class="s">&#39;hbadeals_output.txt&#39;</span><span class="p">,</span><span class="w"> </span><span class="n">sep</span><span class="o">=</span><span class="s">&#39;\t&#39;</span><span class="p">,</span><span class="n">quote</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">,</span><span class="n">col.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">T</span><span class="p">,</span><span class="n">row.names</span><span class="w"> </span><span class="o">=</span><span class="w"> </span><span class="bp">F</span><span class="p">)</span>
</pre></div>
</div>
<section id="hba-deals-arguments">
<h2>HBA-DEALS arguments<a class="headerlink" href="#hba-deals-arguments" title="Permalink to this heading"></a></h2>
<ul class="simple">
<li><dl class="simple">
<dt>countsData: A table of gene names, transcript names and transcript counts in each sample.  At least two transcripts must correspond to</dt><dd><p>each gene.</p>
</dd>
</dl>
</li>
<li><p>labels:  An ordered vector of 1’s and 2’s.  Its length is ncol(countsdata)-2.  Each entry indicates whether the corresponding sample/column
of countsData belongs to the first experimental condition or the second.</p></li>
<li><p>n.cores: The number of cores to use in the calculation.  It is recommended to dedicate as many cores as possible.</p></li>
<li><p>isoform.level: if <code class="docutils literal notranslate"><span class="pre">true</span></code>, return 1-probability of differential proportion for each transcript be returned. If <code class="docutils literal notranslate"><span class="pre">FALSE</span></code>  (the default) return 1-probability of differential splicing for each gene.</p></li>
<li><p>mcmc.iter: The number of iterations of the MCMC algorithm after warmup.</p></li>
<li><p>mcmc.warmup: The number of warmup iterations of the MCMC algorithm.</p></li>
<li><p>hierarchy:  Determines whether a hierarchical model will be used (hierarchy=’yes’), a flat one (hierarchy=’no’) or will the decision
will be made automatically (hierarchy=’auto’)</p></li>
<li><p>lib.size:  A numeric vector containing total library sizes for each sample. If not provided, the default is columnwise count totals.</p></li>
<li><p>mtc:  A logical argument (default FALSE) that indicates whether the output probabilities should be corrected for multiple comparisons.</p></li>
</ul>
</section>
<section id="output-format">
<h2>Output format<a class="headerlink" href="#output-format" title="Permalink to this heading"></a></h2>
<p>The HBA-DEALS output file contains 4 columns. The first column is the gene name, the second is the transcript name,
the third is the fold change, and the fourth is 1-probability of differential expression or proportion(splicing),
which is the posterior error probability (PEP).  Entries that
refer to expression have ‘Expression’ in their second column.  If isoform.level is FALSE, entries that refer to differential
splicing of the gene will have ‘Splicing’ in their second column entry.  The fold change for expression is
given as log2 fold change, and for splicing as fold change.</p>
<table class="docutils align-default">
<thead>
<tr class="row-odd"><th class="head"><p>Gene</p></th>
<th class="head"><p>Isoform</p></th>
<th class="head"><p>ExplogFC/FC</p></th>
<th class="head"><p>P</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>ENSG00000000419</p></td>
<td><p>Expression</p></td>
<td><p>0.00360856329234098</p></td>
<td><p>0.98668</p></td>
</tr>
<tr class="row-odd"><td><p>ENSG0000000041</p></td>
<td><p>ENST00000371588</p></td>
<td><p>0.996357449743301</p></td>
<td><p>0.95506</p></td>
</tr>
<tr class="row-even"><td><p>ENSG00000196141</p></td>
<td><p>Expression</p></td>
<td><p>-2.74059622402359</p></td>
<td><p>0</p></td>
</tr>
<tr class="row-odd"><td><p>ENSG00000196233</p></td>
<td><p>ENST00000371103</p></td>
<td><p>0.10984085543658</p></td>
<td><p>0.04972</p></td>
</tr>
</tbody>
</table>
</section>
</section>


           </div>
          </div>
          <footer><div class="rst-footer-buttons" role="navigation" aria-label="Footer">
        <a href="transcriptome.html" class="btn btn-neutral float-left" title="Inferring transcript readcounts" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>&#169; Copyright 2022.</p>
  </div>

  Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    provided by <a href="https://readthedocs.org">Read the Docs</a>.
   

</footer>
        </div>
      </div>
    </section>
  </div>
  <script>
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script> 

</body>
</html>